packages/nimble-components/src/rich-text/mention-list-box/index.ts

```diff
@@ -0,0 +1,294 @@
+
+    private suggestionProps!: SuggestionProps;
+
+    private readonly intersectionObserver: IntersectionObserver = new IntersectionObserver(
+        entries => {
+            if (
+                !entries[0]?.isIntersecting
+            ) {
+                this.setOpen(false);
+            }
+        },
+        { threshold: 1.0, root: document }
+    );
+
+    /**
+     * @public
+     */
+    public async selectFirstOptionIfValidOptionExists(): Promise<void> {
+        const definedElements = [
+            this.listBox?.matches(':not(:defined)')
+                ? customElements.whenDefined(this.listBox.localName)
+                : Promise.resolve()
+        ];
+        await Promise.all(definedElements);
+        this.updateSelectableOption();
+        if (this.hasAnySelectableOption) {
+            this.listBox?.selectFirstOption();
+            this.scrollOptionIntoView();
+        }
+    }
+
+    /**
+     * @public
+     */
+    public keydownHandler(event: KeyboardEvent): boolean {
+        if (!this.open) {
+            return false;
+        }
+        switch (event.key) {
+            case keyTab:
+            case keyEnter: {
+                if (
+                    this.listBox.firstSelectedOption
+                    && this.hasAnySelectableOption
+                ) {
+                    this.activateMention({
+                        href: this.listBox.firstSelectedOption.value,
+                        displayName: this.listBox.firstSelectedOption.innerText
+                    } as MentionDetail);
+                    return true;
+                }
+                return false;
+            }
+            case keyArrowDown: {
+                if (!event.shiftKey && this.hasAnySelectableOption) {
+                    this.listBox.selectNextOption();
+                    this.scrollOptionIntoView();
+                    return true;
+                }
+                return false;
+            }
+            case keyArrowUp: {
+                if (!event.shiftKey && this.hasAnySelectableOption) {
+                    this.listBox.selectPreviousOption();
+                    this.scrollOptionIntoView();
+                    return true;
+                }
+                return false;
+            }
+            case keyEscape: {
+                this.setOpen(false);
+                return false;
+            }
+            default: {
+                return false;
+            }
+        }
+    }
+
+    /**
+     * @public
+     */
+    public clickHandler(e: MouseEvent): boolean {
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+        const captured = (e.target as HTMLElement).closest(
+            'option,[role=option]'
+        ) as ListOption | null;
+
+        if (!captured || captured.disabled) {
+            return false;
+        }
+        this.activateMention({
+            href: captured.value,
+            displayName: captured.innerText
```

Comment:
Use `text` instead of `innerText`. It has cleaned up whitespace and internally uses `textContent`, which is more efficient than `innerText`.

Reply:
Updated the logic